scripts/street_sign_recognizer.py

```diff
@@ -16,49 +17,144 @@ class StreetSignRecognizer(object):
     def __init__(self):
         """ Initialize the street sign reocgnizer """
         rospy.init_node('street_sign_recognizer')
-        self.cv_image = None                        # the latest image from the camera
+        self.cv_image = None                       # the latest image from the camera
+        self.hsv_image = None
+        self.image_info_window = 255*np.ones((500,500,3))
         self.bridge = CvBridge()                    # used to convert ROS messages to OpenCV
         cv2.namedWindow('video_window')
+        cv2.setMouseCallback('video_window', self.process_mouse_event)
         rospy.Subscriber("/camera/image_raw", Image, self.process_image)
 
+        # hsv slider
+        cv2.namedWindow('threshold_image')
+        self.hsv_lb = np.array([0, 0, 0]) # hsv lower bound
+        cv2.createTrackbar('H lb', 'threshold_image', 0, 255, self.set_h_lb)
+        cv2.createTrackbar('S lb', 'threshold_image', 0, 255, self.set_s_lb)
+        cv2.createTrackbar('V lb', 'threshold_image', 0, 255, self.set_v_lb)
+        self.hsv_ub = np.array([255, 255, 255]) # hsv upper bound
+        cv2.createTrackbar('H ub', 'threshold_image', 0, 255, self.set_h_ub)
+        cv2.createTrackbar('S ub', 'threshold_image', 0, 255, self.set_s_ub)
+        cv2.createTrackbar('V ub', 'threshold_image', 0, 255, self.set_v_ub)        
+
+
     def process_image(self, msg):
         """ Process image messages from ROS and stash them in an attribute
             called cv_image for subsequent processing """
         self.cv_image = self.bridge.imgmsg_to_cv2(msg, desired_encoding="bgr8")
+        self.hsv_image = cv2.cvtColor(self.cv_image, cv2.COLOR_BGR2HSV)
+
+        # lb = (self.hsv_lb[0], self.hsv_lb[1], self.hsv_lb[2])
+        # ub = (self.hsv_ub[0], self.hsv_ub[1], self.hsv_ub[2])
+        lb = (20,170,165)
+        ub = (30,255,255)
+        self.binary_image = cv2.inRange(self.hsv_image, lb, ub)
+
 
         left_top, right_bottom = self.sign_bounding_box()
         left, top = left_top
         right, bottom = right_bottom
 
         # crop bounding box region of interest
-        cropped_sign = self.cv_image[top:bottom, left:right]
+        cropped_sign = self.binary_image[top:bottom, left:right]
 
         # draw bounding box rectangle
         cv2.rectangle(self.cv_image, left_top, right_bottom, color=(0, 0, 255), thickness=5)
 
+    
+    def process_mouse_event(self, event, x,y,flags,param):
+        """ Process mouse events so that you can see the color values associated
+        with a particular pixel in the camera images """
+        self.image_info_window = 255*np.ones((500,500,3))
```

Comment:
Just noticed that you set self.image_info_window here and in in the __init__. Probably unnecessary.  (I assume this is just hanging code from some debuggin)